provider/queue.go

```diff
@@ -36,19 +37,29 @@ func NewQueue(ctx context.Context, name string, ds datastore.Datastore) (*Queue,
 	if err != nil {
 		return nil, err
 	}
+	cancelCtx, cancel := context.WithCancel(ctx)
 	q := &Queue{
 		name:    name,
-		ctx:     ctx,
+		ctx:     cancelCtx,
 		head:    head,
 		tail:    tail,
 		ds:      namespaced,
 		dequeue: make(chan cid.Cid),
 		enqueue: make(chan cid.Cid),
+		close:   cancel,
 	}
 	q.work()
 	return q, nil
 }
 
+// Close stops the queue
+func (q *Queue) Close() error {
+	log.Error("Before close")
+	q.close()
```

Comment:
We may want to wait for the queue to actually close (e.g., have some channel to signal that it's done).